keps/0031-namespace-management.md

```diff
@@ -0,0 +1,167 @@
+
+ * [Summary](#summary)
+ * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+ * [Namespace Management in the Eco-System](#namespace-management-in-the-eco-system)
+    * [Kubernetes Objects and Namespaces](#kubernetes-objects-and-namespaces)
+    * [Helm](#helm)
+    * [The Many Meanings of a Namespace](#the-many-meanings-of-a-namespace)
+    * [Hierarchical Namespace Controller](#hierarchical-namespace-controller)
+ * [Proposals](#proposals)
+    * [Proposal: Single Namespace Support](#proposal-single-namespace-support)
+    * [Proposal: Namespace Creation](#proposal-namespace-creation)
+    * [Proposal: Multi-Namespace (static)](#proposal-multi-namespace-static)
+    * [Proposal: Dependencies and Creating Namespaces](#proposal-dependencies-and-creating-namespaces)
+    * [Proposal: Namespace Cleanup](#proposal-namespace-cleanup)
+ * [Notes](#notes)
+ * [Alternatives](#alternatives)
+    * [Proposal: No namespace creation](#proposal-no-namespace-creation)
+
+## Summary
+
+This KEP aims to provide a definition on namespace management and provide guidance and constraints a KUDO operator may use namespaces.  This includes the expectations of creation, deletion, defining how an operator will install to a namespace and the ability of an operator to leverage more than one namespace.
+
+## Motivation
+
+Like many Kubernetes Object, there is an expectation that a KUDO operator is organized by namespace.  This KEP defines expectations around namespace creation and use.
+
+### Goals
+
+In order to support a multi-namespace environoment, it is necessary to support cluster-wide resources (including kudo instnaces) which is in part (or full) defined in [KEP-05](0005-cluster-resources-for-crds.md).  Based on this constraint, the goals are limited to the existing limitations in KUDO of having namespace only support for instances and not having cluster-wide resource support.
+
+* Define if a namespace is a prerequistie to operator installation
+* Define if a namespace is created by KUDO
+* Define support for advance namespaces (namespaces with metadata)
+* Define if an operator can be installed into a new namespace
+
+### Non-Goals
+
+* multi-namespace support for an operator
+* support for an operator to create or manage a namespace
+
+## Namespace Management in the Eco-System
+
+### Kubernetes Objects and Namespaces
+
+When working kubernetes objects, be it Pods, ReplicaSets, Deployments, etc. that require a namespace, the responsbility falls on the user to create that namespace prior assigning the resource or it will fail.  Available is an [auto-provision admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#namespaceautoprovision) available to auto-create namespaces. Users desiring auto-provisioning of namespaces should install this tool.  This webhook has the limitation of creating namespaces without extra metadata which defeats the value for use cases which need that metadata.
```

Comment:
```suggestion
When working kubernetes objects, be it Pods, ReplicaSets, Deployments, etc. that require a namespace, the responsbility falls on the user to create that namespace prior assigning the resource or it will fail.  Available is an [auto-provision admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#namespaceautoprovision) to auto-create namespaces. Users desiring auto-provisioning of namespaces should install this tool.  This webhook has the limitation of creating namespaces without extra metadata which defeats the value for use cases which need that metadata.
```